### ISSUE TITLE
Debugging apps

### ISSUE THREAD

**Issue by bolt-blue** [jbendtsen/tiny-android-template#3]
I'm trying to figure out a clean way

1. to produce debug builds
2. to debug them

### tl;dr

**Do you have any suggestions for these?**

If you need more info on my setup, let me know. What follows covers what I've been trying thus far.

### Note on my use of your template

I frankensteined a copy of this template using the kotlin `MainActivity.kt` along with the `JniExample.c` and the `androidx` resources.

I have (seemingly) successfully managed to compile a signed `.apk` following your `README`.

### Debug builds

At present, I very naively and probably incorrectly just added `android:debuggable='true'` to `AndroidManifest.xml`.

I also added `-g -O0` to `jni-compile.sh`.

This feels a little janky, but once I understand what needs doing, I can tweak the scripts to have a clean option. 

### Debugging

I presume we should be able to attach `gdb` or `lldb` to a running process; or to maybe use android studio's apk debugging facilities.

For the former, I added `-D` to the `am` command in `run.sh`.
Doing so yields a "Waiting for debugger to be attached" message, but when I manage to connect `gdb` (for better or worse, via [this](https://stackoverflow.com/a/53735039) approach), the message remains, with only its option to 'Force quit'.
I believe this may be due to attaching a native code debugger, but not a java debugger. Not sure.

For the latter, android studio gives this error, but can't really find more detail:

```
Error while starting native debug session: com.intellij.execution.ExecutionException: Unsupported device. This device cannot be debugged using the native debugger. See log file for details.
```

Part of my immediate problem seems related to me testing using an android emulator.
I've had to initially recompile for `x86_64` instead of the default `arm` variant, but that's perhaps not enough.
